Sources/TuistKit/Utils/XCResultParser.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import XCResultKit
+
+final class XCResultParser {
+    private let resultFile: XCResultFile
+    private let invocationRecord: ActionsInvocationRecord
+
+    init?(url: URL) {
+        let resultFile = XCResultFile(url: url)
+        guard let invocationRecord = resultFile.getInvocationRecord() else { return nil }
+        self.resultFile = resultFile
+        self.invocationRecord = invocationRecord
+    }
+
+    func successfulTestTargets() -> Set<String> {
```

Comment:
This can:
- either be an extension of the `invocationRecord` type
- or let's rename `XCResultParser` to `XCResultService` since with this method, it's not just a parser.